modules/integration/tests-ui-integration/src/test/java/org/wso2/carbon/mb/ui/test/dlc/DLCDurableTopicTestCase.java

```diff
@@ -0,0 +1,284 @@
+
+        // Creating a initial JMS consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig = new
+                AndesJMSConsumerClientConfiguration(ExchangeType.TOPIC, DLC_TEST_DURABLE_TOPIC);
+        // Amount of message to receive
+        consumerConfig.setDurable(true, DLC_TEST_DURABLE_TOPIC);
+        consumerConfig.setSubscriptionID("durable-topic-sub-1");
+        consumerConfig.setMaximumMessagesToReceived(EXPECTED_COUNT + 200L);
+        consumerConfig.setAcknowledgeMode(JMSAcknowledgeMode.CLIENT_ACKNOWLEDGE);
+        consumerConfig.setAcknowledgeAfterEachMessageCount(EXPECTED_COUNT + 200L);
+
+        AndesJMSPublisherClientConfiguration publisherConfig =
+                new AndesJMSPublisherClientConfiguration(ExchangeType.TOPIC, DLC_TEST_DURABLE_TOPIC);
+        publisherConfig.setNumberOfMessagesToSend(SEND_COUNT);
+
+        consumerClient = new AndesClient(consumerConfig, true);
+        consumerClient.startClient();
+
+        AndesClient publisherClient = new AndesClient(publisherConfig, true);
+        publisherClient.startClient();
+
+        waitUntilMessagesReceived(6L);
+
+        //Thread sleep until messages sent to DLC after breaching maximum number of retrying
+        AndesClientUtils.sleepForInterval(80000L);
+
+    }
+
+
+    /**
+     * This test will verify delete and restore functions in dlc for durable topic messages.
+     * 1. delete first element in dlc table. if message id doesn't exist in dlc ui table after
+     * delete test will be success.
+     * 2. restore first element in dlc table. if consumer client receive new messages after restore
+     * from dlc test will be success.
+     * TODO: After reroute feature completed in ui there should be a test for reroute durable topic messages.
+     *
+     * @throws java.io.IOException
+     */
+    @Test()
+    public void performDurableTopicDeadLetterChannelTestCase() throws IOException {
+
+        String deletingMessageID;
+        String restoringMessageID;
+
+        DLCBrowsePage dlcBrowsePage = homePage.getDLCBrowsePage();
+        Assert.assertNotNull(dlcBrowsePage.isDLCCreated(),
+                             "DeadLetter Channel not created. " + DLC_TEST_DURABLE_TOPIC);
+        //Testing delete messages
+        DLCContentPage dlcContentPage = dlcBrowsePage.getDLCContent();
+        deletingMessageID = dlcContentPage.deleteFunction();
+
+        Assert.assertTrue(checkMessages(deletingMessageID, DLC_TEST_DURABLE_TOPIC),
+                          "Deleting messages of dead letter channel is unsuccessful.");
+
+        // number of messages received by consumer client before restore function triggered.
+        Long beforeRestoreMessageReceivedCount = consumerClient.getReceivedMessageCount();
+
+        //Testing restore messages
+        restoringMessageID = dlcContentPage.restoreFunction();
+        waitUntilMessagesReceived(9L);
```

Comment:
In case we need such a method for later as well we can break the 'waitForMessagesAndShutdown' method into two. One method which only 'waitForMessages', another method named 'waitForMessagesAndShutdown' which calls the previous method 'waitForMessages' 'and on return, calls shutdown as well.